Implement ebreak properly

Risheng1128 opened the pull request:
This commit adds the instruction `c.ebreak` into rv32emu.

Input the following command, and the `c.ebreak ` test will pass.
> make arch-test RISCV_DEVICE=C

jserv commented on src/emulate.c:
```diff
@@ -148,6 +148,37 @@ static void rv_except_illegal_insn(struct riscv_t *rv, uint32_t insn)
     rv->csr_mcause = code;
 }
 
+void rv_except_breakpoint(struct riscv_t *rv)
```
For the sake of consistency, the function prototype should be
```c
void rv_except_breakpoint(struct riscv_t *rv, uint32_t insn UNUSED);
```

jserv reviewed (changes requested):
Change `README.md` to reflect the results of RISC-V Architecture Tests.

jserv commented:
Check the article [How to Write a Git Commit Message](https://cbea.ms/git-commit/) to summarize this change.

Risheng1128 committed:
Implement c.ebreak properly

Add the breakpoint exception handler which triggers
a breakpoint exception when the c.ebreak instruction
is executed.

Risheng1128 commented:
For the consistent prototype with other exception handlers. I redeclare the `rv_except_breakpoint`.
```diff
- void rv_except_breakpoint(struct riscv_t *rv)
+ static void rv_except_breakpoint(struct riscv_t *rv, uint32_t old_pc)
```

Add the new function `breakpoint_handler` and change `README.md`.

jserv commented on src/riscv.h:
```diff
@@ -125,6 +125,9 @@ riscv_word_t rv_get_reg(struct riscv_t *, uint32_t reg);
 void syscall_handler(struct riscv_t *rv);
 
+/* breakpoint exception handler */
+void breakpoint_handler(struct riscv_t *rv);
```
In order to differentiate breakpoints in the GDB stub, rename to `ebreak_handler`.

jserv commented:
Check `privilege` item in RISC-V Architecture Tests as well.

Risheng1128 commented:
`privilege` instuction test result: 
```shell
Check ebreak                    ... FAIL 
Check ecall                     ... FAIL 
Check misalign1-jalr-01         ... OK 
Check misalign2-jalr-01         ... OK 
Check misalign-beq-01           ... OK 
Check misalign-bge-01           ... OK 
Check misalign-bgeu-01          ... OK 
Check misalign-blt-01           ... OK 
Check misalign-bltu-01          ... OK 
Check misalign-bne-01           ... OK 
Check misalign-jal-01           ... OK 
Check misalign-lh-01            ... OK 
Check misalign-lhu-01           ... OK 
Check misalign-lw-01            ... FAIL 
Check misalign-sh-01            ... OK 
Check misalign-sw-01            ... FAIL
```

jserv commented:
> `privilege` instruction test result:
> Check ebreak                    ... FAIL 
> Check ecall                     ... FAIL 

Create a new issue regarding `privilege` instruction regressions.

Risheng1128 commented:
Rename the function:
```diff
- void breakpoint_handler(struct riscv_t *rv);
+ void ebreak_handler(struct riscv_t *rv);
```

jserv commented:
Thank @Risheng1128 for contributing!
